tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -778,6 +922,79 @@ int main(int argc, char **argv)
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&wire_identities_in, 0));
+
+            struct s2n_connection *conn = s2n_connection_new(S2N_SERVER);
+            EXPECT_NOT_NULL(conn);
+            EXPECT_OK(s2n_connection_set_psk_type(conn, S2N_PSK_TYPE_RESUMPTION));
+
+            struct s2n_config *config_with_cb = s2n_config_new();
+            EXPECT_NOT_NULL(config_with_cb);
+            uint16_t expected_wire_choice = 0;
+            EXPECT_SUCCESS(s2n_config_set_psk_selection_callback(config_with_cb, s2n_test_select_psk_identity_callback, &expected_wire_choice));
+            EXPECT_SUCCESS(s2n_setup_ticket_key(config_with_cb));
+
+            struct s2n_config *config = s2n_config_new();
+            EXPECT_NOT_NULL(config);
+            EXPECT_SUCCESS(s2n_setup_ticket_key(config));
+
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+
+            conn->actual_protocol_version = S2N_TLS13;
+            conn->secure.cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+
+            DEFER_CLEANUP(struct s2n_stuffer psk_identity = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&psk_identity, 0));
+            EXPECT_OK(s2n_setup_encrypted_ticket(conn, &psk_identity));
+
+            /* Resumption psk was selected */
+            {
+                EXPECT_OK(s2n_write_test_identity(&wire_identities_in, &psk_identity.blob));
+
+                EXPECT_OK(s2n_client_psk_recv_identity_list(conn, &wire_identities_in));
+
+                EXPECT_EQUAL(conn->psk_params.chosen_psk_wire_index, 0);
+                EXPECT_NOT_NULL(conn->psk_params.chosen_psk);
+            }
+            
+            /* Customer selects a valid resumption psk with a callback */
+            {
+                conn->psk_params.chosen_psk = NULL;
+                EXPECT_SUCCESS(s2n_stuffer_rewrite(&wire_identities_in));
+                EXPECT_SUCCESS(s2n_connection_set_config(conn, config_with_cb));
+                EXPECT_OK(s2n_write_test_identity(&wire_identities_in, &psk_identity.blob));
+
+                EXPECT_OK(s2n_client_psk_recv_identity_list(conn, &wire_identities_in));
+
+                EXPECT_EQUAL(conn->psk_params.chosen_psk_wire_index, 0);
+                EXPECT_NOT_NULL(conn->psk_params.chosen_psk);
+
+            }
+
+            /* Customer selects an invalid resumption psk with a callback */
+            {
+                conn->psk_params.chosen_psk = NULL;
+                EXPECT_SUCCESS(s2n_stuffer_rewrite(&wire_identities_in));
+                EXPECT_SUCCESS(s2n_connection_set_config(conn, config_with_cb));
+
+                struct s2n_blob wire_identity = { 0 };
+                EXPECT_SUCCESS(s2n_blob_init(&wire_identity, test_bytes_data_2, sizeof(test_bytes_data_2)));
+                EXPECT_OK(s2n_write_test_identity(&wire_identities_in, &wire_identity));
+
+                EXPECT_ERROR(s2n_client_psk_recv_identity_list(conn, &wire_identities_in));
```

Comment:
We could assert for the error code returned here by using `EXPECT_ERROR_WITH_ERRNO` macro